### DISCUSSION TITLE
All Updates In One PR

### DISCUSSION THREAD

**Discussion by sHermanGriffiths** [renovatebot/renovate#32699] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

From the documentation, it seems like Renovate will create multiple PRs by default (e.g. one for updating package1, one for package2, one for patching a vulnerability, one for lockfile management, etc.). Is there any option or combination of options that will ensure that one PR contains all changes Renovate might make? To get specific, I'd like to have one GitHub PR at the start of my day every Monday to review the changes Renovate wants to make. If at all possible, I'd also prefer to have separate commits for each change that would've been a separate PR (see the "e.g." above). This is my current config file (NOTE: I'm just now adding Renovate to the repo so it hasn't run yet):
```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": ["config:recommended"],
    "timezone": "America/Chicago",
    "schedule": ["* 0-7 * * 1"],
    "osvVulnerabilityAlerts": true,
    "lockFileMaintenance": { "enabled": true },
    "customManagers": [
        {
            "customType": "regex",
            "fileMatch": ["requirements\\.apt$"],
            "matchStrings": [
                "#\\s*Renovate start:[\\s\\S]*?#\\s*Renovate stop:",
                "(?<depName>[^=]+)=(?<currentValue>\\S+)"
            ],
            "matchStringsStrategy": "recursive",
            "versioningTemplate": "loose",
            "datasourceTemplate": "deb"
        }
    ],
    "packageRules": [
        {
            "groupName": "All Dependency Updates",
            "matchPackageNames": ["/.+/"],
            "rangeStrategy": "pin",
            "enabled": true
        },
        {
            "matchPackageNames": ["/^python$/"],
            "enabled": false
        }
    ]
}
```
As you can see from the first package rule, I'm under the impression that package rules with a specified `groupName` will all be grouped into one PR and that I can ensure that all package names are matched using a regex wildcard. Am I misguided and if so how do I modify this configuration to get the aforementioned desired result? Any comments about other misconfigurations you might notice are, of course, welcome as well 😊!

### Logs (if relevant)

<details><summary>Logs</summary>

```

I have no logs to post, as Renovate has yet to run on my repo.

```

</details>

**Comment by rarkins** [comment, marked as answer]
A preset already exists: https://docs.renovatebot.com/presets-group/#groupallnonmajor

(Trust me you do not want to group all major updates together)

Renovate creates one commit per branch though, not one commit per dependency/update

**Reply by sHermanGriffiths** [reply]
Thanks for the advice. Should I also not group the other types together (i.e. `pin`, `pinDigest`, `digest`, etc.)?

**Reply by sHermanGriffiths** [reply]
Actually, never mind. I can think of some cases where I would want those separated out as well. Thanks again!

**Reply by rarkins** [reply]
Pinning is a one time event so best to do it separately

**Reply by ghost** [reply]
Is there any way to get renovate to create a separate commit per update in 1 PR versus grouping all in 1 commit?

**Comment by palfrey** [comment]
The answer partially solves things, but there's still an issue with `lockFileMaintenance`. As far as I can tell, (at least with Poetry) if you try to merge lock file updates with everything else it just fails to do lock file maintenance (I suspect because of https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/poetry/artifacts.ts#L175)

I could just only do lockfile maintenance (and implicitly get the other updates) but I'm guessing that'll break with things like github actions updates where they don't have lockfiles. Thoughts?
